### PR TITLE
Update for julia v0.5

### PR DESCRIPTION
**Pull request by yeesian**
- use GeoInterface
- drop support for julia0.4, and support julia0.5 onwards

### CODE REVIEW

**Comment by yeesian** [comment]
depends on https://github.com/JuliaGeo/GeoInterface.jl/pull/8

File: src/GeoJSON.jl [hunk 1]
```diff
@@ -2,20 +2,121 @@ __precompile__()
...
-        FeatureCollection,
-        geojson
-
-    include("types.jl")
-    include("parser.jl")
+    import GeoInterface
+    import JSON
+
+    export parse,
+           parsefile,
```
Comment (visr) [inline src/GeoJSON.jl]: `parse` is a Base function that should probably not be exported. I see that both `parse` and `parsefile` exist in JSON but both not exported.

Comment (yeesian) [inline src/GeoJSON.jl]: 👌

**Comment by yeesian** [comment]
@visr no hurry; but this is now ready for review

**Comment by visr** [comment]
working on it

**Comment by visr** [comment]
Nice, I have a commit pending that updates the readme.

**Comment by yeesian** [comment]
Sorry for adding one more change that came up as I was working on https://github.com/yeesian/LeafletJS.jl. Here are some samples: http://nbviewer.jupyter.org/github/yeesian/LeafletExamples.jl/blob/master/notebook/leaflet-providers.ipynb

Without the PR, there would be issues with string quotes inside nested objects when using `JSON.json()`, and the API is now more consistent with naming conventions:

- `dict2geo()` takes a `Dict` and converts it into a `GeoInterface.*` object
- `geo2dict()` takes a `GeoInterface.*` object and converts it into a `Dict`
- `GeoJSON.geojson()` returns a `String` (obeying the GeoJSON spec) similar to what `JSON.json()` might return.

**Comment by visr** [comment]
Added a few more commits. Go ahead and merge if you agree with those.

**Comment by yeesian** [comment]
Thanks!

**Pull request merged**